processor/lsmintervalprocessor/processor.go

```diff
@@ -313,28 +328,19 @@ func (p *Processor) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) erro
 
 	for _, k := range keys {
 		if err := p.batch.Merge(k, vb, nil); err != nil {
-			errs = append(errs, fmt.Errorf("failed to merge to db: %w", err))
+			return fmt.Errorf("failed to merge to db: %w", err)
 		}
 	}
 
 	if p.batch.Len() >= dbCommitThresholdBytes {
 		if err := p.batch.Commit(p.wOpts); err != nil {
```

Comment:
[**For reviewers**] I initially wanted to move the commit and close code outside hot path by creating a channel and  reading from it in the `Start` loop, however, that would increase the complexity of draining the channel before harvest as well as orchestrating a bit on when to drain the channel. Additionally, a no-buffer channel the performance gains were not substantial for now. We could still come back to this in the future for currently I think that optimization is not required.

Here is the branch with a rough draft of the approach for anyone interested: https://github.com/elastic/opentelemetry-collector-components/compare/main...lahsivjar:opentelemetry-collector-components:poc-get-commit-outside-hotpath?expand=1